### ISSUE TITLE
nuitka and libpng

### ISSUE THREAD

**Issue by Starbuck5** [pygame-community/pygame-ce#2954]
Received a report from a user on discord that png loading in their game crashes when frozen with Nuitka

Windows, Py3.10, pygame-ce 2.5.0, "latest nuitka"

https://discord.com/channels/772505616680878080/822265298085347368/1254516405492256949

Opening this as a placeholder so it doesn't get lost, seems like Andrew has already investigated.

**Comment by oddbookworm** [comment]
According to the tool nuitka uses to determine binary dependencies, here's what imageext depends on upstream
![image](https://github.com/pygame-community/pygame-ce/assets/49015102/d2fe5cdb-ae8d-41f1-b58f-4b147319af1d)

Here's what imageext depends on for pygame-ce
![image](https://github.com/pygame-community/pygame-ce/assets/49015102/e02f28d3-386a-4ec9-935e-eab5d24aa09f)

**Comment by oddbookworm** [comment]
For now, I've opened a pull request with Nuitka

**Comment by Starbuck5** [comment]
I believe this issue would also occur with other video and audio formats, beyond just libpng and libjpeg.

**Comment by oddbookworm** [comment]
I tried with audio and fonts, and they seemed fine

**Comment by oddbookworm** [comment]
[reproducer.zip](https://github.com/user-attachments/files/15950083/reproducer.zip)

Here's the sample project I was working with
Build command:
```
nuitka --standalone main.py --include-data-files=./alien1.png=alien1.png --include-data-files=./alien1.jpg=alien1.jpg --include-data-files=./Roboto-Regular.ttf=Roboto-Regular.ttf --include-data-files=./boom.wav=boom.wav

**Comment by Starbuck5** [comment]
SDL_image additional DLLs:
![image](https://github.com/pygame-community/pygame-ce/assets/46412508/f352ec9a-3943-4502-af4c-5e7cb79d652e)

SDL_mixer additional DLLs:
![image](https://github.com/pygame-community/pygame-ce/assets/46412508/2e174ba2-5608-4cd2-93f4-6e85d503a786)

We don't bundle 100% of these (filesize concerns), but we bundle most of them.

Can't we just tell nuitka if it's a DLL and it's in our directory it's important?

**Comment by oddbookworm** [comment]
I tried a wildcard `*.dll`, and that didn't work

**Comment by Starbuck5** [comment]
Also we have to share this "hook" space with pygame?

**Comment by oddbookworm** [comment]
Yep, because our stuff is in the `pygame` folder

**Comment by oddbookworm** [comment]
*insert plug for briefcase, which just works*

**Comment by Starbuck5** [comment]
>  I tried with audio and fonts, and they seemed fine

You tried with a WAV file. It would have the same issue as the original reporter if you tried opus or mp3 or anything else.

>  insert plug for briefcase, which just works

Let's have all the tools work!

**Comment by oddbookworm** [comment]
I've updated the pull request to Nuitka to do a wildcard glob of all dlls in the pygame(-ce) install. I don't like doing that, but it seems to be our only option with the way the sdl dynamic linking works. libpng doesn't work without the nuitka fix for us, but does work for upstream because of our custom SDL_image build. Other dlls are broken for both of us, such as libtiff and libjpeg

**Comment by Starbuck5** [comment]
I feel like nuitka should do a wildcard glob of DLLs by default. What’s wrong with including all the executables?

We’re not ever going to ship out executables in our wheels that aren’t actually used.

**Comment by oddbookworm** [comment]
> I feel like nuitka should do a wildcard glob of DLLs by default. What’s wrong with including all the executables?
> 
> We’re not ever going to ship out executables in our wheels that aren’t actually used.

This has already been basically answered in the nuitka pull request, but for future reference:
It's basically nuitka trying to save on executable size by not including things that aren't necessary. All of our dlls are necessary, but I wouldn't be surprised if there are quite a few projects out there which include things that aren't really necessary

**Comment by oddbookworm** [comment]
Going to wait to close this until Nuitka 2.4.8 comes out, which I've been promised will have the pull request incorporated which fixes the issue
https://github.com/Nuitka/Nuitka/pull/2947

**Comment by Starbuck5** [comment]
@oddbookworm I'm assuming the issue is fixed and this can be closed?

**Comment by Starbuck5** [comment]
Yes, I tested using the reproducer script Andrew posted on the nuitka PR.
